### PR TITLE
NavigationView for DetailView

### PR DESCRIPTION
**Pull request by evancrow**
### How was this tested?
- [x] I tested this manually
- [ ] I added automated tests
- [x] Existing automated tests are enough to cover my changes

### Screenshots and screen recordings
https://user-images.githubusercontent.com/37356984/157173449-128f6c4c-0634-4ea3-8696-7a46bfe1e317.MP4

## Issues addressed
Fixed: #3009

### CODE REVIEW

File: Client/Frontend/Browser/Card/CardModels.swift [hunk 1]
```diff
@@ -660,6 +659,10 @@ class SpaceCardModel: CardModel {
                 Defaults[.seenBlackFridayNotifyPromo] = true
             })
     }
+
+    func openedSpace() {
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/CardModels.swift:661]: not needed?

File: Client/Frontend/Browser/Card/SpaceDetailList.swift [hunk 2]
```diff
@@ -74,182 +74,180 @@ struct SpaceDetailList: View {
...
+                    if let space = space, !space.isDigest {
+                        SpaceCommentsView(space: space, model: spaceCommentsModel)
+                            .modifier(ListSeparatorModifier())
+                            .id("CommentSection")
                     }
-                    .useEffect(deps: spaceCommentsModel.addingComment) { addingComment in
-                        self.addingComment = addingComment
+                }
+                .modifier(ListStyleModifier(isDigest: space?.isDigest ?? false))
+                .navigationBarTitle("Space")
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/SpaceDetailList.swift]: why do we need this?

File: Shared/API/SpaceStore+CoreSpotlight.swift [hunk 3]
```diff
@@ -74,7 +76,7 @@ extension SpaceStore {
     }
 
     /// Remove spaces from corespoltight by id
-    class func removeSpacesFromCoreSpotlight(
+    public class func removeSpacesFromCoreSpotlight(
```
Comment (yozuysal) [inline Shared/API/SpaceStore+CoreSpotlight.swift:79]: I am not sure why we need this change in this PR

File: Client/Frontend/Browser/Card/CardGrid.swift [hunk 4]
```diff
@@ -92,7 +92,24 @@ struct CardGrid: View {
...
+    }
+
+    @ViewBuilder
+    var detailedSpaceView: some View {
+        if let detailedSpace = detailedSpace {
+            DetailView(primitive: detailedSpace)
+                .onDisappear {
+                    gridModel.closeDetailView()
+                }
+        } else {
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/CardGrid.swift:106]: we shouldnt need the else here, if with a ViewBuilder should give and empty view when the condition is not met

**Comment by evancrow** [comment]
Found a few more crashes I need to fix...

File: Client/Frontend/Browser/Card/GridModel.swift [hunk 5]
```diff
@@ -24,7 +24,7 @@ class GridModel: ObservableObject {
         }
     }
 
-    @Published var showingDetailView = false
+    @Published var showingDetailView = true
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/GridModel.swift:27]: tbh I am not really sure why we need this state at all. This should always be equivalent to spaceModel.detailedSpace != nil, and keeping it as a separate state may leave them prone to getting out of sync. Hmm, I wonder if there is a way to not need this at all.

**Review by yozuysal — COMMENTED** [review]
Thanks for the changes! Test drove it and it actually looks pretty good!

There are a few things about the state handling that I think we will need to cleanup, but tbh I am torn between taking this change as is and over the course of the week slowly tweaking the state handling logic vs changing all in place. It is quite a large PR.. Let's chat tomorrow.

**Review by yozuysal — COMMENTED** [review]
thanks for these changes, but if you have zoomed in on the crash itself, I would make this only about the crash. As I mentioned, I think we will need to think through the state dependencies a bit before we make a call here.

Right now this slices through the isShowingDetails logic (which may be a good way) but it does it by adding a circular dependency to spaceModel. I would suggest reverting these changes and only addressing unfollow crash.

**Comment by evancrow** [comment]
Unfortunately, the state change was my fix to the crash :(. I did add an optional option to `manager` which avoided the crash but broke the functionality.

**Comment by yozuysal** [comment]
Umm, frankly that sounds like an overkill for a crash to unfollow.. there should be a way to bandaid fix that functionality without changing detail showing logic for all spaces. (Again, I think that logic should change, but this is gonna be in a rush)

**Comment by evancrow** [comment]
Yeah haha, was a bit overkill. Pretty much put the logic back to the way it was, just changed how detailedSpace was set.

**Comment by yozuysal** [comment]
not sure what changed with Fixes for iPad, but now the drag from the edge to close is not working on phones.

**Comment by evancrow** [comment]
This is what you would see on iPad when the app launched before my fix:

File: Client/Frontend/Browser/Card/CardGrid.swift [hunk 6]
```diff
@@ -148,100 +151,28 @@ struct CardGrid: View {
...
-                }.modifier(
-                    DraggableDetail(
-                        detailDragOffset: $detailDragOffset,
-                        width: geom.size.width))
+                NavigationLink(
+                    destination: detailedSpaceView,
+                    isActive: $gridModel.showingDetailView
+                ) {}.useEffect(deps: spaceModel.detailedSpace) { detailedSpace in
+                    gridModel.showingDetailView = detailedSpace != nil
+                }
```
Comment (yozuysal) [inline Client/Frontend/Browser/Card/CardGrid.swift:159]: I think this is missing .environment(\.onOpenURLForSpace), we are falling back to openURL which loses setting the parentID and rootID to spaceID

Comment (yozuysal) [reply Client/Frontend/Browser/Card/CardGrid.swift:159]: also the shareURL is missing so share button crashes.

I think we pretty much need to carry over all important environemnt variables :)

Comment (evancrow) [reply Client/Frontend/Browser/Card/CardGrid.swift:159]: I follow that logic :)

Comment (evancrow) [reply Client/Frontend/Browser/Card/CardGrid.swift:159]: `onOpenURLForSpace` is passed down via BrowserView (as is `openURL`). `ShareURL` was missing but I just added it, will push in a sec.

Comment (yozuysal) [reply Client/Frontend/Browser/Card/CardGrid.swift:159]: onOpenURLForSpace is definitely not getting through though. When you click any link, look for the back button being enabled, if it is not, then there was no onOpenURLForSpace and it fell back to use openURL instead.

**Review by yozuysal — APPROVED** [review]
I think this is in good shape now. Thanks Evan! 🤞

**Comment by evancrow** [comment]
Tests not related to this and failing from main, going to merge

**Pull request merged**